src/lsm/scan_groove.zig

```diff
@@ -0,0 +1,797 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+const ScanTreeType = @import("scan_tree.zig").ScanTreeType;
+const ScanBufferType = @import("scan_buffer.zig").ScanBufferType;
+const GridType = @import("grid.zig").GridType;
+
+const Direction = @import("direction.zig").Direction;
+const TimestampRange = @import("timestamp_range.zig").TimestampRange;
+
+pub fn ScanGrooveType(
+    comptime Groove: type,
+    comptime Storage: type,
+    comptime scan_max: comptime_int,
+) type {
+    comptime assert(std.meta.fields(Groove.IndexTrees).len > 0);
+
+    return struct {
+        const Self = @This();
+        const ScanBuffer = ScanBufferType(Storage);
+
+        pub const Scan = ScanType(Groove, Storage, scan_max);
+        pub const Fetcher = FetcherType(Scan, Groove, Storage);
+
+        slots: [scan_max]Scan = undefined,
```

Comment:
Thinking about what `scans_max` should be, I realized that we must use the `Dispatcher` union only for `ScanTreeType` implementations, as it makes sense to limit `scans_max` because they consume memory and io.
The set operations `ScanMerge{}Type`s can have their own limit defined as `scans_max / 2`.